ravensproject/ravensproject/Agent.java

```diff
@@ -112,10 +112,37 @@ private void CompareObjects(HashMap<String, RavensObject> aObject, HashMap<Strin
 			System.out.println(attr_b);
 		}
 		
-		if(attr_a.keySet().size() == 1)
+		for(String s : cObject.keySet())
```

Comment:
How about create a method that accepts this hashmap and returns the attr ?
Also, you don't have to extract all the keyset if you don't need to.. you could add that in the loop.